charts/kyverno/templates/_helpers.tpl

```diff
@@ -122,3 +122,15 @@ maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
 {{ toYaml .Values.securityContext }}
 {{- end }}
 {{- end }}
+
+{{- define "kyverno.imagePullSecret" }}
+{{- with index . 0 }}
+{{- printf "{\"auths\":{\"%s\":{\"auth\":\"%s\"}}}" .registry (printf "%s:%s" .username .password | b64enc) | b64enc }}
+{{- end }}
+{{- end }}
+
+{{- define "kyverno.imagePullSecretNames" -}}
```

Comment:
The template could take care of the trailing `,`. 
It looks strange that the caller has to account for it.

Reply:
Now that the actual value is a map, I found simpler way that doesn't really warrant a definition. So now just join the keys from map in deployment template.

Reply:
✅